### PR TITLE
refactor: simplify checkContainsTrigger by constructing regex once and optimize regex compilation

### PR DESCRIPTION
**Pull request by nishantm77**
This PR cleans up a few code smells in `checkContainsTrigger`:
1. It moves the regex compilation for `triggerPhrase` to the top of the function so that it is only constructed once per invocation, rather than rebuilding it inside multiple `if` blocks.
2. It removes the identical ternary branches in the issue/PR comment check where both the true and false conditions returned `context.payload.comment.body`.
**Changes made**
* Hoisted `const regex = new RegExp(...)` to the beginning of the `checkContainsTrigger` function.
* Replaced the identical ternary branch with a standard `context.payload.comment.body || ""` assignment.

### CODE REVIEW

**Comment by nishantm77** [comment]
@copilot review. I've updated the regex construction so that if triggerPhrase is empty, it falls back to a non-matching regex /(?!)/. This completely prevents the overly permissive behavior while keeping the rest of the logic intact.

File: src/github/validation/trigger.ts [hunk 1]
```diff
@@ -22,6 +22,12 @@ export function checkContainsTrigger(context: ParsedGitHubContext): boolean {
     return true;
   }
 
+  // Compile the regex once for checking the trigger phrase
+  // If triggerPhrase is empty, use a regex that never matches anything
+  const regex = triggerPhrase
+    ? new RegExp(`(^|\\s)${escapeRegExp(triggerPhrase)}([\\s.,!?;:]|$)`, "i")
+    : /(?!)/;
```
[the remark this answers is not in the document]
Comment (nishantm77) [reply src/github/validation/trigger.ts]: I've also added a unit test for the empty triggerPhrase behavior in test/trigger-validation.test.ts. @copilot please review again.

File: src/github/validation/trigger.ts [hunk 2]
```diff
@@ -22,6 +22,12 @@ export function checkContainsTrigger(context: ParsedGitHubContext): boolean {
...
+
```
[the remark this answers is not in the document]
Comment (nishantm77) [reply src/github/validation/trigger.ts]: I've moved the regex compilation down so it only runs if the event bypasses the assignee and label early-returns, avoiding the performance penalty. @copilot please review again.
